doc/man3/flux_future_create.adoc

```diff
@@ -0,0 +1,110 @@
+flux_future_create(3)
+=====================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_create, flux_future_destroy, flux_future_get, flux_future_check, flux_future_then, flux_future_set_timeout - synchronize an activity
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef struct flux_future_struct flux_future_t;
+
+ typedef void (*flux_continuation_f)(flux_future_t *f, void *arg);
+
+ flux_future_t *flux_future_create (flux_reactor_t *r);
+
+ void flux_future_destroy (flux_future_t *f);
+
+ int flux_future_set_timeout (flux_future_t *f, double timeout);
+
+ int flux_future_get (flux_future_t *f, void *result);
+
+ bool flux_future_check (flux_future_t *f);
+
+ int flux_future_then (flux_future_t *f, flux_continuation_f cb, void *arg);
+
+
+DESCRIPTION
+-----------
+
+A Flux future represents some activity that may be completed with reactor
+watchers and/or message handlers.  It is intended to be wrapped
+by other classes that perform specific activities, providing a common idiom
+for synchronizing on the result.  Authors of classes that wrap
+futures should refer to `flux_future_set_init(3)`.
+
+`flux_future_create()` creates a future and associates it with a
+Flux reactor.  A class that uses futures typically wraps this in
+a class-specific method returning a `flux_future_t`.  A future is
+destroyed with `flux_future_destroy()`.
+
+`flux_future_check()` returns true if a call to `flux_future_get()`
```

Comment:
I think the intended use case is to decide whether to register the continuation, not repeated polling.   The man page should no doubt recommend against repeated polling.